### PR TITLE
Debug pipeline

### PR DESCRIPTION
**Pull request by paulbaron**
-Spot light fixed
-Configurable draw line
-Frustum of spotlights calculated and drawn

### CODE REVIEW

File: GameEngine/GameEngine/Engine/Systems/RenderCameraSystem.cpp [hunk 1]
```diff
@@ -69,11 +70,42 @@ namespace AGE
...
 
-			// WARNINNNNNNNNNNNG
-			// Le calcul n'est pas bon !!!!
-			// Paul, fais le STP, merci
+			float spotFov = glm::max(0.001f, (1.0f - spot->getCutOff()) * glm::radians(180.0f));
+//			// We add 30 degres to have the spot entierly in the frustum (otherwise it is circumscribed)
+			float adaptedFov = glm::min(spotFov + glm::radians(30.0f), glm::radians(179.9f));
+			glm::mat4 invTransform = glm::inverse(spotEntity->getLink().getGlobalTransform());
+			glm::mat4 spotViewProj = glm::perspective(adaptedFov, 1.0f, 0.1f, 1000.0f) * invTransform;
+
```
Comment (DorianPinaud) [inline GameEngine/GameEngine/Engine/Systems/RenderCameraSystem.cpp]: Il l'a fait !!!!! :dancer:

File: GameEngine/GameEngine/Engine/Systems/RenderCameraSystem.cpp [hunk 2]
```diff
@@ -69,11 +70,42 @@ namespace AGE
...
+			glm::vec3 color = glm::vec3(1, 0, 0);
+			bool activateDepth = true;
+
+			AGE::GetRenderThread()->getQueue()->emplaceCommand<Commands::ToRender::Draw3DQuad>(aNear, bNear, cNear, dNear, color, activateDepth);
+			AGE::GetRenderThread()->getQueue()->emplaceCommand<Commands::ToRender::Draw3DQuad>(aFar, bFar, cFar, dFar, color, activateDepth);
+			AGE::GetRenderThread()->getQueue()->emplaceCommand<Commands::ToRender::Draw3DLine>(aNear, aFar, color, activateDepth);
+			AGE::GetRenderThread()->getQueue()->emplaceCommand<Commands::ToRender::Draw3DLine>(bNear, bFar, color, activateDepth);
+			AGE::GetRenderThread()->getQueue()->emplaceCommand<Commands::ToRender::Draw3DLine>(cNear, cFar, color, activateDepth);
+			AGE::GetRenderThread()->getQueue()->emplaceCommand<Commands::ToRender::Draw3DLine>(dNear, dFar, color, activateDepth);
```
Comment (DorianPinaud) [inline GameEngine/GameEngine/Engine/Systems/RenderCameraSystem.cpp]: Très styler ! :)

**Comment by DorianPinaud** [comment]
good for me !

File: GameEngine/GameEngine/Engine/Utils/ThreadName.hpp [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 #ifdef WIN32
 
+#define NOMINMAX
```
Comment (cesarl) [inline GameEngine/GameEngine/Engine/Utils/ThreadName.hpp]: What does that mean ?

Comment (paulbaron) [inline GameEngine/GameEngine/Engine/Utils/ThreadName.hpp]: I dont know why but when I updated glm, the only problem I had was that glm::min and glm::max where in conflict with the min and max functions from windows (it shouldn't as it is in the glm namespace...) so this define just removes the min and max functions from windows.h (we dont use them anyway).

**Pull request merged**